### PR TITLE
:sparkles: delete full node suggestion

### LINKED ISSUE #119: Delete full party tree suggestion

### PR DESCRIPTION
**Pull request by nothingelsematters**
Closes #119. It isn't possible to test buttons anyway

### CODE REVIEW

**Comment by nothingelsematters** [comment]
Here's a job for ya, mate:

- [message text](https://github.com/pool-party/pull-party-bot/blob/93237b3ac9012f3f18b691b47690ae7d3c8cb6ee/src/main/kotlin/com/github/pool_party/pull_party_bot/commands/handlers/DeleteCommands.kt#L66)
- [button text](https://github.com/pool-party/pull-party-bot/blob/93237b3ac9012f3f18b691b47690ae7d3c8cb6ee/src/main/kotlin/com/github/pool_party/pull_party_bot/commands/handlers/DeleteCommands.kt#L68)
- [what to send here](https://github.com/pool-party/pull-party-bot/blob/93237b3ac9012f3f18b691b47690ae7d3c8cb6ee/src/main/kotlin/com/github/pool_party/pull_party_bot/commands/handlers/callback/DeleteNodeSuggestionCallback.kt#L32)

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/handlers/callback/DeleteNodeSuggestionCallback.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        if (!validateAdministrator(callbackQuery.from, message.chat, false)) {
+            answerCallbackQuery(
+                callbackQuery.id,
+                ON_PERMISSION_DENY_CALLBACK
+            )
+            return
+        }
+
+        if (partyDao.deleteNode(partyId)) {
+            answerCallbackQuery(callbackQuery.id, onPartyDeleteSuccess("TODO"))
```
Comment (ShuffleZZZ) [inline src/main/kotlin/com/github/pool_party/pull_party_bot/commands/handlers/callback/DeleteNodeSuggestionCallback.kt:33]: Would be nice to have a behavior similar to what we have in `partyDao.delete` (to have a party or even a list of deleted aliases as return value)

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt [hunk 2]
```diff
@@ -4,17 +4,33 @@ val ON_CLEAR_SUCCESS =
...
+fun onAliasDeleteSuccess(partyName: String) = """Alias $partyName is no longer exists 👍"""
+
+fun onPartyDeleteSuccess(partyName: String) = """Party $partyName and its aliases are deleted 👍"""
+
+fun onPartyDeleteSuggest(partyList: List<String>): String {
+    val message = """Perhaps you want to delete all aliases of"""
+
+    val format =
+        when (partyList.size) {
+            0 -> """"""// TODO check if case is possible.
```
Comment (nothingelsematters) [inline src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt]: huh?

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt [hunk 3]
```diff
@@ -4,17 +4,33 @@ val ON_CLEAR_SUCCESS =
...
+            1 -> """ ${partyList.last()}"""
```
Comment (nothingelsematters) [inline src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt:24]: triple quotes are redundant, aren't they?

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt (hunk 2 above)
Comment (ShuffleZZZ) [reply src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt]: I were not sure, can be removed if there is a contract of nonempty list

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt (hunk 3 above)
Comment (ShuffleZZZ) [reply src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt:24]: Trying to keep style the same. don't mind going back to the single quotes

Comment (nothingelsematters) [reply src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt:24]: I think, triple quotes should be used only for multiline strings 🙃

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt (hunk 2 above)
Comment (nothingelsematters) [reply src/main/kotlin/com/github/pool_party/pull_party_bot/commands/messages/DeleteMessages.kt]: Well, there is a contract that if the last alias is deleted, then the party itself is deleted, as well. So there cannot be an empty party list, afaik 🙃

**Comment by ShuffleZZZ** [comment]
Can we test this or just merge?

**Comment by nothingelsematters** [comment]
> Can we test this or just merge?

merge, kekw

**Comment by nothingelsematters** [comment]
> Can we test this or just merge?

users will test

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/Handlers.kt [hunk 4]
```diff
@@ -27,7 +27,7 @@ fun Bot.initHandlers() {
 
     val chatDaoImpl = ChatDaoImpl()
 
-    val callbacks = listOf(RemoveSuggestionCallback(partyDaoImpl), PingCallback(partyDaoImpl))
+    val callbacks = listOf(DeleteSuggestionCallback(partyDaoImpl), PingCallback(partyDaoImpl))
```
Comment (ShuffleZZZ) [inline src/main/kotlin/com/github/pool_party/pull_party_bot/commands/Handlers.kt:30]: Shouldn't there be all 3 callbacks?

File: src/main/kotlin/com/github/pool_party/pull_party_bot/commands/handlers/callback/DeleteNodeSuggestionCallback.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import com.elbekD.bot.Bot
+import com.elbekD.bot.types.CallbackQuery
+import com.github.pool_party.pull_party_bot.commands.Callback
+import com.github.pool_party.pull_party_bot.commands.CallbackAction
+import com.github.pool_party.pull_party_bot.commands.messages.ON_PARTY_DELETE_SUCCESS
+import com.github.pool_party.pull_party_bot.commands.messages.ON_PERMISSION_DENY_CALLBACK
+import com.github.pool_party.pull_party_bot.commands.validateAdministrator
+import com.github.pool_party.pull_party_bot.database.dao.PartyDao
+
+class DeleteNodeSuggestionCallback(private val partyDao: PartyDao) : Callback {
```
Comment (ShuffleZZZ) [inline src/main/kotlin/com/github/pool_party/pull_party_bot/commands/handlers/callback/DeleteNodeSuggestionCallback.kt:12]: Feels like there is a lot of common code with `DeleteSuggestionCallback` class, you can reformat it if you have a will

**Review by ShuffleZZZ — DISMISSED** [review]

**Review by ShuffleZZZ — APPROVED** [review]

**Pull request merged**